src/features/treasureIsland/components/SandPlot.tsx

```diff
@@ -0,0 +1,358 @@
+      gameState.context.state.inventory["Sand Shovel"]?.gte(1);
+
+    if (!hasSandShovel) {
+      handleNoShovel();
+      return;
+    }
+
+    gameService.send("REVEAL", {
+      event: {
+        type: "treasure.dug",
+        id,
+        createdAt: new Date(),
+      },
+    });
+
+    sandPlotSend("DIG");
+  };
+
+  const acknowledgeTreasureFound = () => {
+    if (!sandPlotState.context.discovered) return;
+
+    setToast({
+      icon: ITEM_DETAILS[sandPlotState.context.discovered].image,
+      content: `+1`,
+    });
+
+    sandPlotSend("ACKNOWLEDGE");
+  };
+
+  const acknowledgeNoSandShovel = () => {
+    setShowMissingShovelModal(false);
+    onMissingShovelAcknowledge();
+  };
+
+  const handleTreasureCheck = () => {
+    // Each time the sprite sheet gets to the 10th frame (shovel up)
+    // If reward has returned then stop sprite here.
+    if (reward !== undefined) {
+      goblinDiggingRef.current?.pause();
+      setShowGoblinEmotion(true);
+
+      if (!reward.discovered) {
+        // If no reward is found then move the sandPlot machine into
+        // finished state. No user interaction required.
+        setTimeout(() => {
+          sandPlotSend({
+            type: "FINISH_DIGGING",
+            discovered: reward.discovered,
+            dugAt: reward?.dugAt,
+          });
+        }, 2000);
+      }
+    }
+  };
+
+  if (sandPlotState.matches("dug")) {
+    return (
+      <div className="w-full h-full">
+        <img
+          src={SUNNYSIDE.soil.sand_dug}
+          className="absolute"
+          style={{
+            width: `${PIXEL_SCALE * 16}px`,
+            top: `${PIXEL_SCALE * 2}px`,
+          }}
+        />
+      </div>
+    );
+  }
+
+  if (sandPlotState.matches("treasureFound")) {
+    return (
+      <Modal centered show onHide={acknowledgeTreasureFound}>
+        <CloseButtonPanel onClose={acknowledgeTreasureFound}>
+          <Revealed onAcknowledged={acknowledgeTreasureFound} />
+        </CloseButtonPanel>
+      </Modal>
+    );
+  }
+
+  if (showMissingShovelModal) {
+    return (
+      <Modal centered show onHide={acknowledgeNoSandShovel}>
+        <CloseButtonPanel
+          title="No Sand Shovel!"
+          onClose={acknowledgeNoSandShovel}
+        >
+          <div className="p-2 pt-0 mb-2 flex flex-col items-center space-y-2">
+            <img
+              src={ITEM_DETAILS["Sand Shovel"].image}
+              alt="Sand Shovel"
+              onLoad={(e) => setImageWidth(e.currentTarget)}
+            />
+            <p>
+              You need to have a Sand Shovel equipped to be able to dig for
+              treasure!
+            </p>
+            <p>
+              If you need to purchase one, you can head to the Treasure Shop at
+              the southern end of the island.
+            </p>
+          </div>
+          <Button onClick={acknowledgeNoSandShovel}>Got it</Button>
+        </CloseButtonPanel>
+      </Modal>
+    );
+  }
+
+  const isDigging =
```

Comment:
Why can't we use `sandPlotState.matches("digging")`? it sounds like that state indicates the Goblin must be digging

Reply:
We need to be able to see the goblin AFTER the action of digging is complete. Its the time when he has finished digging but pausing for the reveal of the reward.